### DISCUSSION TITLE
QR code on desktop web + "AR button" on mobile web --> SceneViewer or QuickLook

### DISCUSSION THREAD

**Discussion by JMAR09** [google/model-viewer#2933] [Q&A]
Hi guys,

Is there any option how to generate QR code on desktop web for the users (in the same way how it is possible in Model Viewer editor)? The idea is to do not have Model Viewer on desktop and to have just QR code which will send users to SceneViewer or QuickLook. 

And very similar thing on mobile web but instead of QR code to have "AR button" which will send users to SceneViewer or QuickLook. 

Is it possible to do it in this way? 

Thank you very much

Jan 

![image](https://user-images.githubusercontent.com/91330743/139302514-4530eeb2-62b3-43bd-8d36-57b52030c94a.png)

**Comment by elalish** [comment]
Yes, it's possible, though if you  don't want to show  the 3D model on the web, then model-viewer is probably overkill for your purposes. You can launch Quick  Look and Scene Viewer directly with a couple lines of  code if that's all you want to do. However, we find users are  much more likely to use AR and accept its  permissions if  they can see and  interact  with the 3D model first on the web. 

The  editor QR code  solution is complicated because  the 3D model is uploaded  by the user. For your case  it  should be  as simple as posting a QR for your mobile URL.

**Comment by JMAR09** [comment]
Hi Emmet, 

thank you for your answer :) 

You are right, that it would be better to do it by Model Viewer. 

On the other hand, I see a potential of "AR corners" in retail shops. Customers could try other products which are not there by scanning QR code. Plus my company has a some issue with implementing Model Viewer to the desktop/mobile web, we will start with the app and this is the main reason, why I am thinking how to generate QR code/AR button for the web. 

I have more questions, is there any chance to call you by Google Meets and ask on them? It would be much more easier for me. My "technical" english is not great, it quite dificult to write it understandable way. If you would have a time, it would be awesome.

Thank you 

Have a great weekend

Jan

**Reply by elalish** [reply]
I'd love to hear about any issues you're having with with  MV on the desktop/mobile  web. But no, I don't think there's time for a call (and it looks like we are in very different time zones anyway). Your english is  just fine; I'm having no trouble understanding it. Give it a shot! 👍

**Reply by JMAR09** [reply]
Hi Emmet, to be honest it is quite difficult for me to tell you where is the problem. E-shop, where I am working, has desktop and mobile version. On desktop web MV is working very well 👉 [https://www.alza.cz/EN/ar-test](https://www.alza.cz/EN/ar-test) and on mobile web it is not working at all 👉 [https://m.alza.cz/EN/ar-test](https://m.alza.cz/EN/ar-test). I am sure that our developers will know how to fix it, but it is not their priority now. I have to wait. 


On the other hand I got developers time at least for our app 👉 [https://www.alza.cz/EN/aplikacie-alzask-a-alza-media-art7565.htm](https://www.alza.cz/EN/aplikacie-alzask-a-alza-media-art7565.htm). The goal is to implement Model Viewer + implement AR option for customers by SceneViewer and QuickLook. 

Based on previous two articles, I have these questions/ideas:

1. Do I need something special to launch SceneViewer, when I have file in GLB or glTF? For example with USDZ file it is enough to download and QuickLook just starts automatically.
2. Would it be difficult for you to add "QR code" to the Model Viewer on desktop? From my point of you it is pity, when users are on desktop, they can see "just" 3D model and now sign that it is possible to see in AR view. My idea is this:

- QR code could be generate it somewhere else and then just implement like picture to the code
- On the desktop web MV could be button for QR code
- It could realy speed up the process for guys like me (I have very low HTML/CSS knowledge), there will be no block on desktop. I tried something on my old traveling blog and I really missed this QR code option for desktop 👉 [https://vocasontheway.cz/ar-test-2/](https://vocasontheway.cz/ar-test-2/)
- I believe it could be doable for you

![image](https://user-images.githubusercontent.com/91330743/139596764-a3780c99-e628-40c5-9015-2e31a3971a3d.png)

3. As I wrote above, my coding skills are not great, but still I can do at least something. I tried to follow MV documentation+examples, I was able to implement something, but I am not able to implement everything. I believe that there are more people like me. What could help similar guys like me? 

- Webinar/course how to work with MV
- Interactive code in "MV example", which could have effect on MV window (I expect this could be very difficult to do it from your side) 

![image](https://user-images.githubusercontent.com/91330743/139596898-afdcf443-119e-4a8a-a6ed-f5cb980ae48b.png)

I hope that everything is understandable and I believe that it could help Model Viewer to be more simple tool for not "technical" people. 

I am really looking forward to read your answer. 

Thank you very much 

Kind Regards

Jan

P.s.: I am from Prague, CET time zone.

**Reply by elalish** [reply]
1) No, launching Scene Viewer is pretty simple; just use their documented intent, passing in the GLB URL.
2) This doesn't really make sense inside our library, as there are lots of ways to do this already pretty easily. In fact Chrome even has some features built in to help you open a tab from your desktop on your mobile. Nothing is stopping you from putting a QR code on your website that points to itself for a mobile device to scan.
3) The closest we have to interactive code is our [Editor](https://modelviewer.dev/editor/), where you can edit the snippet and see the changes real-time on your model. It doesn't yet support CSS and JS, but all attributes in our API are supported. Otherwise I'd recommend something like Glitch, where you  can remix our [example](https://glitch.com/edit/#!/model-viewer) and see  the results  real time. And yes, some webinars would be great, but we rely on the community for that (you can find a few on YouTube).

**Reply by JMAR09** [reply]
Hi Emmet, thank you for your answers and your time. I will definitely try to do something with Glitch. I did not know it. :)
